src/figuresClasses.ts

```diff
@@ -1,19 +1,77 @@
-export interface Figure {
 
+type Color = 'red' | 'green' | 'blue';
+
+export interface Figure {
+  shape: string;
+  color: Color;
+  getArea: () => number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  public shape: string = 'triangle';
 
-}
+  constructor(
+    public color: Color,
+    private a: number,
+    private b: number,
+    private c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error();
```

Comment:
add error messages everywhere
```suggestion
      throw new Error('error message');
```